src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -751,34 +753,57 @@ def rook_api_patch(self, path: str, **kwargs: Any) -> Any:
     def rook_api_post(self, path: str, **kwargs: Any) -> Any:
         return self.rook_api_call("POST", path, **kwargs)
 
+    def list_storage_classes(self) -> List[str]:
+        try:
+            crd = self.customObjects_api.get_namespaced_custom_object(
+                group="ceph.rook.io",
+                version="v1",
+                namespace=self.rook_env.namespace,
+                plural="cephclusters",
+                name=self.rook_env.cluster_name)
+
+            sc_devicesets = crd['spec']['storage']['storageClassDeviceSets']
+            sc_names = [vct['spec']['storageClassName'] for sc in sc_devicesets for vct in sc['volumeClaimTemplates']]
+            log.info(f"the cluster has the following configured sc: {sc_names}")
+            return sc_names
+        except Exception as e:
+            log.error(f"unable to list storage classes: {e}")
+            return []
+
+    # TODO: remove all the calls to code that uses rook_cluster.storage_class_name
     def get_storage_class(self) -> 'client.V1StorageClass':
```

Comment:
I think this method is not used in any place. Or if it is used, now it should return the list of configured SCs in the cluster deviceset, am i wrong?.
Probably the rook orchestrator option to store the name of the Storage class has no sense with this change.
And this implies that probably it is also not needed at all the `RookCluster` class `storage_class_name` attribute

Reply:
@jmolmo this method is called from:

```
    def add_osds(self, drive_group, matching_hosts):
        # type: (DriveGroupSpec, List[str]) -> str
        assert drive_group.objectstore in ("bluestore", "filestore")
        assert drive_group.service_id
        storage_class = self.get_storage_class()
```
I'm not sure in which context we allow dashboard to add OSDs for a rook environment. AFAIK this is something that rook team wants to avoid because OSDs are controlled by the operator. WDYT?

Reply:
[Drivegroups support was removed from Rook](https://github.com/rook/rook/issues/7275) time ago .. So i think that probably this is not the right way to create osds from the rook orchestrator.
This drives us to think again about the "semantical" and "conceptual" differences between imperative (ceph dashboard) and declarative( rook operator) ways to work.
I will suggest to use the declarative approach , and add new pages to the Ceph Dashboard to manage rook clusters. In my view the right way to add OSDs in a Rook Cluster is changing the cluster CRD.